apps/server/src/modules/tldraw/service/tldraw.ws.service.ts

```diff
@@ -283,6 +294,47 @@ export class TldrawWsService {
 		}
 	}
 
+	private async syncAssets(doc: WsSharedDocDo) {
+		// clean up assets that are not used as shapes anymore
+		// which can happen when users do undo/redo operations on assets
+		const assets: YMap<TldrawAsset> = doc.getMap('assets');
+		const shapes: YMap<TldrawShape> = doc.getMap('shapes');
+		const usedShapesAsAssets: TldrawShape[] = [];
+		const usedAssets: TldrawAsset[] = [];
+
+		shapes.forEach((shape) => {
+			if (shape.assetId) {
+				usedShapesAsAssets.push(shape);
+			}
+		});
+
+		doc.transact(() => {
+			assets.forEach((asset) => {
+				const foundAsset = usedShapesAsAssets.find((shape) => shape.assetId === asset.id);
+				if (!foundAsset) {
+					assets.delete(asset.id);
+				} else {
+					usedAssets.push(asset);
+				}
+			});
+		});
+
+		const files = await this.filesStorageClientAdapterService.listFilesOfParent(doc.name);
```

Comment:
Please rename files in fileRecords...yes the method is also called list*Files*OfParent and should be renamed. But you do not get an file (binary file) only meta informations. Hopefully FileRecords make it more clear for the future.

Reply:
fixed